### PR TITLE
Scaffold webpacker; add ESLint (using `standard`)

### PR DESCRIPTION
**Pull request by bhaibel**
- Using `webpacker` in spite of disliking some of its
  specific choices (cough `yarn` cough) for the sake
  of cleaving to Rails standards; the newest version
  also adds the ability to recompile assets on page
  reload rather than running a separate webpack process
  which should make new-developer onboarding simpler.

- Added `standard` as our base ESLint config because
  I personally like it better than AirBNB or Prettier;
  I care less which linter config we use than that we
  use a prefab one and am open to alternate suggestions.

- Haven't added any specific JS or CSS framework stuff
  here yet as I would specifically like to use a low-
  or no-framework approach on this project. (Faster
  load time => more subscription conversions.)

- I've slung the Webpack stuff in `app/assets` -- this is
  distinct from the default behavior, which is
  `app/javascript`. I find `app/javascript` misleading for
  a directory that will also contain SCSS, image assets,
  etc. I find `app/assets` easier to understand because
  that's where frontend stuff goes in Rails projects, but
  am worried it may mislead or confuse people who are
  expecting to see Sprockets stuff there. Feedback
  specifically requested on this point.

- Apologies for the ginormous PR size; most of it is a
  5.9 KLoC (!!!) `yarn.lock` file.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,15 +13,9 @@ gem "rails", "~> 5.1.4"
...
 gem "puma", "~> 3.7"
-# Use SCSS for stylesheets
-gem "sass-rails", "~> 5.0"
-# Use Uglifier as compressor for JavaScript assets
-gem "uglifier", ">= 1.3.0"
-# See https://github.com/rails/execjs#readme for more supported runtimes
-# gem 'therubyracer', platforms: :ruby
-#
-# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks
-gem "turbolinks", "~> 5"
```
Comment (zspencer) [inline Gemfile]: You had mentioned wanting to keep a pjaxy thing around, I'm assuming webpacker will let us do that in a boring way?

File: app/assets/packs/application.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/* eslint no-console:0 */
+// This file is automatically compiled by Webpack, along with any other files
+// present in this directory. You're encouraged to place your actual application logic in
+// a relevant structure within app/javascript and only use these pack files to reference
+// that code so it'll be compiled.
```
Comment (zspencer) [inline app/assets/packs/application.js]: Is there a way we can not include the fully compiled packages? We'll probably need to figure out a way to make it work on heroku, but committing compiled assets makes all change sets super noisy.

**Comment by zspencer** [comment]
There was a small bug in the setup script that I resolved; but otherwise I think this is good to go, unless you want to think about the two questions I left.

File: Gemfile (hunk 1 above)
Comment (bhaibel) [reply Gemfile]: Yep, gonna add that back in after the scaffold is up

File: app/assets/packs/application.js (hunk 2 above)
Comment (bhaibel) [reply app/assets/packs/application.js]: This is webpacker's comments being confusing -- the COMPILED assets go in `/public/packs`, which is ignored. These files are what webpack calls "entry points" -- when it compiles assets, it uses entry points as the roots of dependency trees that it then compiles into production asset files. So this file is compiled by webpack in the sense that a Java compiler compiles `.java` files, not in the sense that a Java compiler compiles Java bytecode.

Webpacker should work out of the box on heroku as it adds a compilation rake task... I think. That's what it advertises itself as doing!

**Pull request merged**